feat: Application loading ADR (#335)
input-output-hk/hermes: docs/src/architecture/09_architecture_decisions/0004-app-loading.md

---
    title: 0004 Hermes application loading procedure
    adr:
        author: Alex Pozhylenkov <alex.pozhylenkov@iohk.io>
        created: 11-Aug-2024
        status:  draft
---

## Context

> Hermes as an engine which runs user's developed applications,
> obviously should have a functionality to load a and run provided assembled application.
> This procedure must validate application on the integrity and correctness,
> prepare application's state, before actually executing application itself.

Reviewer: This would sound better
```suggestion
Hermes, as an engine that runs user-developed applications, should have the functionality to load and execute the provided assembled application. This process must validate the application's integrity and correctness and prepare the application's state before executing it.
```

## Assumptions

Hermes application package could be corrupted or modified.

## Decision

> During the each application loading and running process application package should be provide
> and all possible validations and initialization should be performed based on the package data.

Reviewer: ```suggestion
During each application loading and running process, the application package should be provided. All possible validations and initialization should be performed based on the package data.
```

## Risks

> * Potential bad UX user experience for each application running,
>   because such validations and state preparation could take some time.

Reviewer: ```suggestion
* Potential bad  user experience (UX) for each application run due to time consumption in validations and state preparation 
```
